WordPressKit/NotificationSyncServiceRemote.swift

```diff
@@ -13,6 +13,10 @@ public class NotificationSyncServiceRemote: ServiceRemoteWordPressComREST {
         case failed
     }
 
+    public enum InputError: Int, Error {
```

Comment:
I wonder if we should just use an `Error` type here? I don't see the purpose of the `Int`. Granted I've not added many of these myself, but a search reveals we use `Error` much more often than `Int, Error`.

Reply:
Because the completion closure accepts a "generic" `Error` type, I needed the `InputError` to be some sort of `RawRepresentable` or `Equatable` to make testable. Indeed from the classes perspective this does not add any value. So it is added for the unit tests. I think we can compare the description in the tests but that didn't seem very clean. Can you think of a neat way of testing this while keeping the type as `InputError: Error`?

Reply:
Ah, I see. That's fine then. It wasn't a big deal anyway, we can leave it as-is. Thanks for the explanation!

Reply:
Cool! I can also build a a overarching Error type over `InputError` & `SyncError` and change the closure argument types from Error to that. But might be an overkill at this case

Reply:
> But might be an overkill at this case

Agreed. 😄